main.py

```diff
@@ -19,144 +31,94 @@ def get_spell_details(spell_url):
-
-                    if not matching_spells:
-                        print(f'No spells found with the name "{spell_name_to_generate}" in the API.')
-                    elif len(matching_spells) == 1:
-                        spell_details = get_spell_details(matching_spells[0]['url'])
-                        if spell_details is not None:
-                            content = create_spell_markdown(spell_details, save_directory)
-                            save_result = save_spell_md_file(spell_details, content, save_directory)
-                            if save_result:
-                                print("MD file created successfully.")
-                    else:
-                        print(f'Multiple spells found with the name "{spell_name_to_generate}":')
-                        for i, spell in enumerate(matching_spells, 1):
-                            print(f"{i}. {spell['name']}")
-
-                        spell_choice = input("Enter the number of the spell you want to generate an MD file for (or type 'back' to return to the main menu): ")
-                        if spell_choice.lower() == 'back':
-                            break  # returns to the main menu
-
-                        try:
-                            spell_index = int(spell_choice) - 1
-                            selected_spell = matching_spells[spell_index]
-                            spell_details = get_spell_details(selected_spell['url'])
-                            if spell_details is not None:
-                                content = create_spell_markdown(spell_details, save_directory)
-                                save_result = save_spell_md_file(spell_details, content, save_directory)
-                                if save_result:
-                                    print("MD file created successfully.")
-                        except (ValueError, IndexError):
-                            print("Invalid input. Please enter a valid number.")
-            elif choice == "3":
-                print("Exiting the program.")
-                break  # stops the program
+
+        # jinja2 template from file
+        template_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), "templates")
+        template_loader = FileSystemLoader(searchpath=template_dir)
+        template_env = Environment(loader=template_loader)
+        spell_template = template_env.get_template('spell_template.md')
+
+        if spell_name_to_generate:
+            matching_spells = [spell for spell in spells if spell_name_to_generate.lower() in spell['name'].lower()]
+
+            if not matching_spells:
+                print(f'No spells found with the name "{spell_name_to_generate}" in the API.')
+            elif len(matching_spells) == 1:
+                spell_details = get_spell_details(matching_spells[0]['url'])
+
+                saving_result(spell_details, spell_template, save_directory)
```

Comment:
the saving_result function returns false/true but you ignore it. is that desirable?